### PR TITLE
DMVP-1164: Weave-scope structure change

### PR DESCRIPTION
**Pull request by VahagnMian**
Changes to Weave-Scope module structure

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -80,3 +80,10 @@ module "external-secrets" {
     module.eks-cluster
   ]
 }
+
+module "weave-scope" {
```
Comment (aramkarapetian) [inline main.tf:84]: Could you please put this into separate file? wave-scope-dashboard.tf maybe

File: modules/weave-scope/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+## Example
+This is an example of usage `weave-scope` module
+
+
+```
+module "weave-scope" {
+  source = "./modules/weave-scope"
```
Comment (aramkarapetian) [inline modules/weave-scope/README.md:7]: Could you please point to terraform registry address?

File: modules/weave-scope/variables.tf [hunk 3]
```diff
@@ -0,0 +1,8 @@
+variable "namespace" {
+  type    = string
+  default = "default"
```
Comment (aramkarapetian) [inline modules/weave-scope/variables.tf:3]: Description please

File: modules/weave-scope/variables.tf [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+}
+variable "create_namespace" {
+  type    = bool
+  default = true
```
Comment (aramkarapetian) [inline modules/weave-scope/variables.tf:7]: Description please

File: modules/weave-scope/variables.tf (hunk 3 above)
Comment (aramkarapetian) [inline modules/weave-scope/variables.tf:3]: I think this should not go into default namespace. meta-system, dasmeta-system?

File: variables.tf [hunk 5]
```diff
@@ -170,3 +170,31 @@ variable "map_roles" {
   }))
   default = []
 }
+
+variable "enable_weave_scope" {
```
[the remark this answers is not in the document]
Comment (aramkarapetian) [reply variables.tf:174]: we should use this variable, not every setup needs the dashboard.

File: variables.tf [hunk 6]
```diff
@@ -170,3 +170,31 @@ variable "map_roles" {
...
+  default     = false
+}
+
+variable "namespace" {
+  type    = string
+  default = "default"
+}
+
+variable "weave_scope" {
+  description = "Additional IAM roles to add to the aws-auth configmap."
```
Comment (aramkarapetian) [inline variables.tf:186]: How is this related to wave scope?

Comment (aramkarapetian) [reply variables.tf:186]: Probably description is wrong or?

File: variables.tf [hunk 7]
```diff
@@ -170,3 +170,31 @@ variable "map_roles" {
...
+  type = list(object({
+    enabled          = bool
+    create_namespace = bool
+    namespace        = string
+  }))
+  default = [
+    {
+      enabled          = false
```
Comment (aramkarapetian) [inline variables.tf:194]: I don't think we need this here, if used along with cluster setup then all those variables will have specific values which do not need to be modified.

File: variables.tf (hunk 5 above)
[the remark this answers is not in the document]
Comment (VahagnMian) [reply variables.tf:174]: Dear Aram, Tigran is right, this variable is moved into weave_scope variable, and can be configured there, so I think we dont need that variable

File: modules/weave-scope/variables.tf (hunk 3 above)
Comment (VahagnMian) [reply modules/weave-scope/variables.tf:3]: gotcha

**Review by mrdntgrn — APPROVED** [review]
